pkg/deploy/certificates/selfsigned_secret.go

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2012-2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package certificates
+
+import (
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/eclipse-che/che-operator/pkg/util"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type SelfSignedSecret struct {
```

Comment:
The name doesn't make sense

Reply:
why?

Reply:
Because it is a certificate that's self-signed, not secret. Secret cannot sign anything.